src/app/shared/components/cart-tooltip/cart-tooltip.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, debounceTime, Subject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CartTooltipService {
+  isVisible$ = new BehaviorSubject(false).pipe(debounceTime(500));
+
+  constructor() {}
+
+  public open() {
```

Comment:
There are two examples of solutions here, like yours (open and close), you could also implement the "toggle(status: boolean)" method, which would be for sense the same as the two methods, but would take up less space in the build in the future, but it's still not even close to critical, but generally there are options like that.